server/etcdserver/server.go

```diff
@@ -1548,6 +1548,10 @@ func (s *EtcdServer) isLearnerReady(id uint64) error {
 	leaderMatch := rs.Progress[leaderID].Match
 	// the learner's Match not caught up with leader yet
 	if float64(learnerMatch) < float64(leaderMatch)*readyPercent {
+		lg.Error(
+			"rejecting promote learner: learner is not ready",
+			zap.String("learner-ready-percent", fmt.Sprint(learnerMatch/leaderMatch*100)),
+		)
```

Comment:
```suggestion
	learnerProgressPercent := float64(learnerMatch) / float64(leaderMatch)
	if learnerProgressPercent < readyPercent {
		lg.Error(
			"rejecting promote learner: learner is not ready",
			zap.Float64("learner-progress-percent", learnerProgressPercent),
			zap.Float64("learner-ready-percent", readyPercent),
		)
```

`learnerMatch/leaderMatch*100` without conversion to float64 will just be equal `0`.

Reply:
> learnerMatch/leaderMatch*100 without conversion to float64 will just be equal 0.

yes, make sense. I missed that

Reply:
> zap.Float64("learner-ready-percent", readyPercent),

this suggestion seems wrong to me as `readyPercent` is not a `learner-ready-percent` 🤔  but `readyPercent` is just a threshold to decide whether to allow a learner to be added or not in cluster.

Reply:
Right, feel free to rename to make it clearer. For example rename `readyPercent` to `readyPercentThreshold` and  add field to log `ready-percent-threshold`.

I think it would be great to return the threshold in the log, so user doesn't need to read the etcd code to find it.

Reply:
done